tutorials/shaders/compute_shaders.rst

```diff
@@ -3,70 +3,159 @@
 Using compute shaders
 =====================
 
+This tutorial will walk you through the process of creating a minimal compute
+shader. But first, a bit of background on compute shaders and how they work with
+Godot.
 
-Think of compute shaders as blocks of code that are executed on the GPU for any purpose we want.
-Compute shaders are independent from the graphics pipeline and do not have much fixed-functionality.
-Contrast this with fragment shaders which are used specifically for assigning a color to a fragment in a render target.
-The big benefit of compute shaders over code executed on a CPU is the high amount of parallelization that GPUs provide.
+.. note::
 
-Because compute shaders are independent of the graphics pipeline we don't have any user defined inputs or outputs
-(like a mesh going into the vertex shader or a texture coming out of a fragment shader). Instead, compute shaders
-make changes directly to memory stored on the GPU from which we can read and write using scripts.
+   This tutorial assumes familiarity with non-compute shaders. If you are new to
```

Comment:
"non-compute shaders" sounds a bit strange. Maybe it should be "vertex and fragment shaders"?

Reply:
What about?

>  This tutorial assumes familiarity with shaders generally.